### PR TITLE
chore: Fix broken TypeScript rewrite script

### PR DESCRIPTION
**Pull request by acdvorak**
This PR fixes the following error when running
`node scripts/rewrite-declaration-statements-for-typescript.js ~/dev/mdc/packages`:

```
[rewrite] packages/mdc-checkbox/constants.ts

node_modules/@babel/parser/lib/index.js:3843
    throw err;
    ^

SyntaxError: This experimental syntax requires enabling the parser plugin: 'classProperties' (56:29)
    at Object.raise (node_modules/@babel/parser/lib/index.js:3831:17)
    at Object.expectPlugin (node_modules/@babel/parser/lib/index.js:5148:18)
    at Object.parseClassProperty (node_modules/@babel/parser/lib/index.js:8229:12)
    at Object.parseClassProperty (node_modules/@babel/parser/lib/index.js:10260:18)
    at Object.pushClassProperty (node_modules/@babel/parser/lib/index.js:8193:30)
    at Object.parseClassMemberWithIsStatic (node_modules/@babel/parser/lib/index.js:8132:14)
    at Object.parseClassMemberWithIsStatic (node_modules/@babel/parser/lib/index.js:10166:11)
    at Object.parseClassMember (node_modules/@babel/parser/lib/index.js:8069:10)
    at Object.parseClassMember (node_modules/@babel/parser/lib/index.js:10123:11)
    at withTopicForbiddingContext (node_modules/@babel/parser/lib/index.js:8024:14)
```

This PR also:

* No longer rewrites `@material/foo` imports (they can be resolved as-is)
* Renames some variables for clarity
* Prints a more helpful error message when no arguments are passed to the script
* Fixes some unrelated Sass stylelint errors
* Updates GitHub API to use non-deprecated authentication method

### CODE REVIEW

File: test/screenshot/infra/lib/github-api.js [hunk 1]
```diff
@@ -43,14 +43,16 @@ class GitHubApi {
 
   constructor() {
     this.gitRepo_ = new GitRepo();
-    this.octokit_ = octokit();
+    this.octokit_ = octokit(this.getAuthToken_());
```
Comment (kfranqueiro) [inline test/screenshot/infra/lib/github-api.js]: Why are there screenshot testing infra changes in this PR?

Comment (acdvorak) [reply test/screenshot/infra/lib/github-api.js]: I regenerated `package-lock.json`, and evidently some of our dependencies don't follow semver, so we got a few breaking changes. I figured I might as well fix them while I'm here.

File: test/unit/mdc-tab-bar/mdc-tab-bar.test.js [hunk 2]
```diff
@@ -227,13 +227,6 @@ test(`#adapter.notifyTabActivated emits the ${MDCTabBarFoundation.strings.TAB_AC
   td.verify(handler(td.matchers.contains({detail: {index: 66}})));
 });
 
-function setupMockFoundationTest(root = getFixture()) {
```
Comment (acdvorak) [inline test/unit/mdc-tab-bar/mdc-tab-bar.test.js]: The function is already defined above, and this copy is missing `const root = ...`.

Comment (kfranqueiro) [reply test/unit/mdc-tab-bar/mdc-tab-bar.test.js]: This copy had it as a parameter instead, but I guess if nothing was ever specifying that parameter it's fine

File: scripts/rewrite-declaration-statements-for-typescript.js [hunk 3]
```diff
@@ -140,72 +149,76 @@ function transform(srcFile, rootDir) {
...
+function isThirdPartyModule(importPath) {
+  // See: https://nodejs.org/api/modules.html#modules_all_together (step 3)
+  const wouldLoadAsFileOrDir = ['./', '/', '../'].some((s) => importPath.indexOf(s) === 0);
+  return !wouldLoadAsFileOrDir;
+}
+
+function getBazelFileNameOrPath(ossFileNameOrPath) {
+  return ossFileNameOrPath
+    .replace(/mdc-?/g, '')
+    .replace(/material-components-web/g, 'material_components_web')
```
Comment (kfranqueiro) [inline scripts/rewrite-declaration-statements-for-typescript.js]: I don't think we even mirror this package?

Comment (acdvorak) [reply scripts/rewrite-declaration-statements-for-typescript.js]: Oh good point. I'll remove it.

Comment (acdvorak) [reply scripts/rewrite-declaration-statements-for-typescript.js]: Removed

**Pull request merged**